code/game/turfs/turf.dm

```diff
@@ -141,6 +141,11 @@
 			if(Obj.flags & PROXMOVE)
 				spawn( 0 )
 					Obj.HasProximity(A, 1)
+
+	if(cameranet.checkTurfVis(src))
+		INVOKE_EVENT(A, /event/cameranet_entered, "target" = A)
+	else
+		INVOKE_EVENT(A, /event/cameranet_exited, "target" = A)
```

Comment:
1. I worry about the performance of checking cameranet visibility on every single movement of every object in the world ever. It looks probably not super expensive, but doing things on movement can add up.
2. These events are invoked on *every single movement*, not just when an object newly enters or exits the cameranet

Reply:
1. limiting this to only mobs and mechas now, since they're the only members of the track list that gets built.
2. can't find any procs that handle that but i'll keep my eyes out.